quantized_mesh_tile/editable_terrain.py

```diff
@@ -0,0 +1,487 @@
+
+    def rebuild_h(self):
+        """
+        Requantize the heights and sets min/max heights of this tile, if heights are
+        changed, otherwise nothing will happens
+        """
+        if self._changed_heights:
+            new_max = max(self._changed_heights)
+            new_min = min(self._changed_heights)
+
+            deniv = new_max - new_min
+            b_height = old_div(MAX, deniv)
+            for i in range(len(self._changed_heights)):
+                changed_height = self._changed_heights[i]
+                h = int(round((changed_height - new_min) * b_height))
+                if h < 0:
+                    h = 0
+                if h > MAX:
+                    h = MAX
+                self.h[i] = h
+
+            self.header['minimumHeight'] = new_min
+            self.header['maximumHeight'] = new_max
+            self._changed_heights = []
+
+    def _rebuild_indices(self):
+        """
+        Private method, should only used internally if any edits
+        on self.u, self.v, self.h  are made.
+        """
+        if self._changed_heights:
+            self.rebuild_h()
+        size_indices = len(self.indices)
+        size_uvh = len(self.u)
+        new_u = [None] * size_uvh
+        new_v = [None] * size_uvh
+        new_h = [None] * size_uvh
+        new_v_light = [None] * size_uvh
+        new_indices = [None] * size_indices
+        index_map = [None] * size_uvh
+
+        new_index = 0
+        for position, old_i in enumerate(self.indices):
+            if index_map[old_i]:
+                (new_i, positions) = index_map[old_i]
+                positions.append(position)
+            else:
+                index_map[old_i] = (new_index, [position])
+                new_index += 1
+
+        for old_i, data in enumerate(index_map):
+            (new_i, positions) = data
+            new_u[new_i] = (self.u[old_i])
+            new_v[new_i] = (self.v[old_i])
+            new_h[new_i] = (self.h[old_i])
+            new_v_light[new_i] = (self.vLight[old_i])
+
+            for position in positions:
+                new_indices[position] = new_i
+
+        if len(self.indices) == len(new_indices):
+            self.indices = new_indices
+        else:
+            raise Exception("Array-Size of Indices not equal")
+
+        self.u = new_u
+        self.v = new_v
+        self.h = new_h
+        self.vLight = new_v_light
+
+        self.westI = self.get_edge_vertices('w')
+        self.southI = self.get_edge_vertices('s')
+        self.eastI = self.get_edge_vertices('e')
+        self.northI = self.get_edge_vertices('n')
+
+    def _quantize_latitude(self, latitude):
+        """
+        Private helper method to convert latitude values to quantized tile (v) values
+        :param latitude: the wgs 84 latitude in degrees
+        :return: the quantized value (v)
+        """
+        b_lat = old_div(MAX, (self._north - self._south))
+        v = int(round((latitude - self._south) * b_lat))
+        return v
```

Comment:
To avoid code duplication we can put this method in TerrainTile and use it there as well. Same comment for the other quantize methods.

Reply:
And dequantize private methods.

Reply:
ok